tests/internal/scripts/ccuda/deallocate.py

```diff
@@ -0,0 +1,19 @@
+from pyccel.decorators import kernel
+from pyccel import cuda
+import math
+
+@kernel
+def func(arr:'int[:]'):
+    """
+    Takes an array 'arr' as input and raise each element to the power 2.
+    """
+    i = cuda.grid(0)
+    arr[i] = math.pow(arr[i], 2)
+
+if __name__ == '__main__':
+    a = cuda.array([1,2,3,4], memory_location='device')
+    b = cuda.array([3,4,5,6], memory_location='host')
+    c = b
```

Comment:
i think it would be better to have the name of the file indicate which free are we targeting with the tests.
then we can know if `cuda_free` or `cuda_free_pinter` is the one that have the error.